code/modules/mob/living/simple_animal/bot/bot_construction.dm

```diff
@@ -563,38 +566,69 @@
-		qdel(I)
+	if(istype(I, /obj/item/robot_parts/l_arm))
+		if(left_arms < 2)
+			left_arms++
+			to_chat(user, "<span class='notice'>You add a left arm to [src]!</span>")
+			qdel(I)
+		else
+			to_chat(user, "<span class='notice'>There are already enough left arms.</span>")
 
-	else if((istype(I, /obj/item/melee/energy/sword)) && (build_step == 3))
-		if(!user.unEquip(I))
-			return
-		to_chat(user, "<span class='notice'>You complete General Griefsky!.</span>")
-		new /mob/living/simple_animal/bot/secbot/griefsky(get_turf(src))
-		qdel(I)
-		qdel(src)
+	else if(istype(I, /obj/item/robot_parts/r_arm))
+		if(right_arms < 2)
+			right_arms++
+			to_chat(user, "<span class='notice'>You add a right arm to [src]!</span>")
+			qdel(I)
+		else
+			to_chat(user, "<span class='notice'>There are already enough right arms.</span>")
 
-	else if((istype(I, /obj/item/toy/sword)) && (toy_step < 3))
-		if(!user.unEquip(I))
-			return
-		toy_step++
-		to_chat(user, "<span class='notice'>You add a toy sword to [src]!.</span>")
-		qdel(I)
+	else if((istype(I, /obj/item/melee/energy/sword)) || (istype(I, /obj/item/toy/sword)))
+		if(right_arms < 2)
+			to_chat(user, "<span class='warning'>[src] needs two right arms before you can start attaching the swords!</span>")
 
-	else if((istype(I, /obj/item/toy/sword)) && (toy_step == 3))
-		if(!user.unEquip(I))
-			return
-		to_chat(user, "<span class='notice'>You complete Genewul Giftskee!.</span>")
-		new /mob/living/simple_animal/bot/secbot/griefsky/toy(get_turf(src))
-		qdel(I)
-		qdel(src)
+		else if(left_arms < 2)
+			to_chat(user, "<span class='warning'>[src] needs two left arms before you can start attaching the swords!</span>")
+
+		else if((istype(I, /obj/item/melee/energy/sword)) && (build_step < 3) && (toy_step == 0))
+			if(!user.unEquip(I))
+				return
+			build_step++
+			to_chat(user, "<span class='notice'>You add an energy sword to [src]!</span>")
+			qdel(I)
+
+		else if((istype(I, /obj/item/melee/energy/sword)) && (toy_step > 0))
+			to_chat(user, "<span class='warning'>You already added a toy sword to [src]! You cannot mix and match different kinds of swords, it would be dangerous.</span>")
+
+		else if((istype(I, /obj/item/melee/energy/sword)) && (build_step == 3))
+			if(!user.unEquip(I))
+				return
+			to_chat(user, "<span class='notice'>You complete General Griefsky!</span>")
+			new /mob/living/simple_animal/bot/secbot/griefsky(get_turf(src))
+			qdel(I)
+			qdel(src)
+
+		else if((istype(I, /obj/item/toy/sword)) && (toy_step < 3) && (build_step == 0))
+			if(!user.unEquip(I))
+				return
+			toy_step++
+			to_chat(user, "<span class='notice'>You add a toy sword to [src]!.</span>")
+			qdel(I)
+
+		else if((istype(I, /obj/item/toy/sword)) && (build_step > 0))
+			to_chat(user, "<span class='warning'>You already added an energy sword to [src]!You cannot mix and match different kinds of swords, it would be inefficient.")
```

Comment:
```suggestion
			to_chat(user, "<span class='warning'>You already added an energy sword to [src]! You cannot mix and match different kinds of swords, it would be inefficient.</span>")
```